### DISCUSSION TITLE
How certificates are generated in courses?

### DISCUSSION THREAD

**Discussion by nidhikachhawa24** [Sunbird-Lern/Community#202] [Documentation]
Hello @AmiableAnil @mohitga @reshmi-nair @muzafferjoya

As I know, by enabling the Issue Certificate while creating a batch can generate certificate for the user.

But I want to understand the full flow the the certificate generation.

Like:
1. Is it generated through telemetry? if yes, what is the cycle of certificate generation?
2. After getting the telemetry completed, what steps needs to be followed to enable certificate?
3. How we can enable the same process of certification generation like courses in EVENTS as well

cc: @komalm @manishakapadnis @paritshivani

**Comment by mohitga** [comment]
Hello @nidhikachhawa24 , 
Please have a look at the mircosite page https://lern.sunbird.org/use/developer-guide/lms-service/lms-flink-jobs/collection-certificate-generator it details architecture of certificate generation. 

Please tag also @maheshkumargangula for lern related queries

**Reply by nidhikachhawa24** [reply]
@mohitga @maheshkumargangula 

I want to generate certificates for events. 

As I checked on BOT : 

> can we attach certificates to EVENTS instead of collection? 
No, the current implementation of the Collection Certificate Generator in Sunbird only supports attaching certificates to collections or courses (course batches). Certificates cannot be attached to individual events.


But if we create batches into the events then certificate generation is possible.

When creating batch for event, I don't get option to attach a certificate template or layout to the batch.

Do we need to update the form for create event for certificates?
How can I attach certificates to EVENTS?

cc: @komalm @manishakapadnis @paritshivani

**Reply by maheshkumargangula** [reply]
@nidhikachhawa24 - Here are the details about the Certificate Generation. 

The certificate generation implemented as a Job which listen to the events and generates the Certificate. The event will have the complete information to issue the certificate.

With the current implementation, the Certificate Generation workflow is enabled for Courses. Batch is created to each course and optionally possible to define a Certificate criteria along with the Certificate Template to issue certificates.

There is "Collection Certificate Preprocessor" which does the evaluation for each enrolment of the Batch (created for Course) and generate an event for "Collection Certificate Generator" to process and issue it. 

The APIs are generalised to enable "Certificate Generation" for various use cases. Please help us understand the use case for which you want to enable certificate generation, what is the Criteria to issue it. 

Cc: @mohitga

**Reply by nidhikachhawa24** [reply]
@maheshkumargangula 

Can we connect in first half? Please let me know suitable time so that we can explain our scenario clearly.

**Reply by maheshkumargangula** [reply]
@nidhikachhawa24 - Let us have a call at 5:30pm today or 12pm on 25th July 2024. Please send an invite.

**Reply by nidhikachhawa24** [reply]
@maheshkumargangula 
Will schedule a meeting today at 5:30.

Can you please share your email so that I can send an invitation?

**Reply by nidhikachhawa24** [reply]
@maheshkumargangula 

Here is the invitation:

Discussion of Cetificates
Tuesday, July 23 · 5:30 – 6:00pm
Time zone: Asia/Kolkata
Google Meet joining info
Video call link: https://meet.google.com/hib-ggap-cek
Or dial: ‪(US) +1 315-754-3520‬ PIN: ‪708 020 114‬#
More phone numbers: https://tel.meet/hib-ggap-cek?pin=7448388278322

**Reply by nidhikachhawa24** [reply]
Hello @maheshkumargangula 

I've rescheduled the meeting at 12Pm on 25th July 2024.

Here is the invitation:

Discussion of Cetificates
Tuesday, July 25 · 12:00 – 12:30pm
Time zone: Asia/Kolkata
Google Meet joining info
Video call link: https://meet.google.com/hib-ggap-cek
Or dial: ‪(US) +1 315-754-3520‬ PIN: ‪708 020 114‬#
More phone numbers: https://tel.meet/hib-ggap-cek?pin=7448388278322

**Comment by paritshivani** [comment]
@mohitga , @maheshkumargangula ,
**is it possible to create/generate certificate bypassing Flink job? or via postman?** 
We have a requirement where we need to provide certificate to other content type than course that is event which have batch also.
Need some guidance here.
